### PR TITLE
feat: Legal Assistant configuration 

### PR DESCRIPTION
**Pull request by Fr4nc3**
Here's the detailed information for the Pull Request:

## Purpose

* Introduce the Legal Assistant feature to the Chat With Your Data (CWYD) platform. This feature includes:
  - Legal documents sample data to help test and demonstrate the legal assistant capabilities.
  - An admin dropdown to select either the default prompt or the legal assistant prompt for specific use cases.
  - A comprehensive README for the Legal Assistant, detailing its purpose, how to update prompt configurations, and question and answer examples.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
# Run the application and navigate to the admin panel to select the Legal Assistant prompt
# Upload sample legal documents and test various queries as per the README guidelines
# Verify the responses are accurate and formatted correctly as per the enhanced prompt
```

## What to Check
Verify that the following are valid:
* The admin dropdown for selecting the Legal Assistant prompt is functional.
* The Legal Assistant responds correctly to the example questions provided in the README.
* The document listings, summaries, and specific clause extractions are accurate and properly formatted.
* The enhanced prompt handles state, business type, date range, and other criteria correctly.
* Ensure the citation format is correctly followed and duplicates are removed.

## Other Information

* Ensure that the Legal Assistant README is detailed enough for new users to understand the setup and functionality without external guidance.
* Double-check the integration with the existing CWYD framework to ensure no conflicts or performance issues.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -426,3 +426,7 @@ tests/integration/ui/cypress/screenshots/
 
 #JetBrains IDE
 .idea/
+package.json
+package-lock.json
+package.json
+package.json
```
Comment (hunterjam) [inline .gitignore:432]: Please remove lines 429 through 432. These should not be in the .gitignore file.

File: code/backend/batch/utilities/helpers/config/config_helper.py [hunk 2]
```diff
@@ -84,6 +84,8 @@ def get_available_loading_strategies(self):
 
     def get_available_orchestration_strategies(self):
         return [c.value for c in OrchestrationStrategy]
+    def get_available_ai_assistant_types(self):
+        return ["default", "legal assistant"]
```
Comment (hunterjam) [inline code/backend/batch/utilities/helpers/config/config_helper.py:88]: Instead of hard-coding values, use an enum. See OrchestrationStrategy as an example.

File: code/backend/batch/utilities/helpers/config/config_helper.py [hunk 3]
```diff
@@ -133,6 +136,7 @@ def _set_new_config_properties(config: dict, default_config: dict):
         Function used to set newer properties that will not be present in older configs.
         The function mutates the config object.
         """
+        print("config", config)
```
Comment (hunterjam) [inline code/backend/batch/utilities/helpers/config/config_helper.py:139]: Remove "print" to console reference.

File: .gitignore [hunk 4]
```diff
@@ -426,3 +426,7 @@ tests/integration/ui/cypress/screenshots/
 
 #JetBrains IDE
 .idea/
+package.json
```
Comment (ross-p-smith) [inline .gitignore:429]: Why are these excluded and multiple times?

File: code/backend/batch/utilities/helpers/config/config_helper.py [hunk 5]
```diff
@@ -184,6 +191,12 @@ def get_active_config_or_default():
 
         return Config(config)
 
+    @staticmethod
+    @functools.cache
+    def get_default_assistant_prompt():
+        config = ConfigHelper.get_default_config()
+        return  config["prompts"]["answering_user_prompt"]
```
Comment (hunterjam) [inline code/backend/batch/utilities/helpers/config/config_helper.py:198]: Should this be: return config.get("answering_user_prompt")?

File: code/backend/batch/utilities/helpers/config/config_helper.py (hunk 3 above)
Comment (ross-p-smith) [inline code/backend/batch/utilities/helpers/config/config_helper.py:139]: Is this debug?

File: code/backend/pages/04_Configuration.py [hunk 6]
```diff
@@ -174,7 +184,7 @@ def validate_documents():
     post_answering_prompt_help = "You can configure a post prompt that allows to fact-check or process the answer, given the sources, question and answer. This prompt needs to return `True` or `False`."
     use_on_your_data_format_help = "Whether to use a similar prompt format to Azure OpenAI On Your Data, including separate system and user messages, and a few-shot example."
     post_answering_filter_help = "The message that is returned to the user, when the post-answering prompt returns."
-
+    ai_assistant_type_help = "Whether to use the default user prompt or other."
```
Comment (hunterjam) [inline code/backend/pages/04_Configuration.py:187]: Since there are only two options, I would rephrase this to "Whether to use the default user prompt or the Legal Assistance user prompt. Refer to the Legal Assistance README for more details."

**Comment by ross-p-smith** [comment]
Hi @Fr4nc3 - this PR looks like it is failing as the Code Coverage decreases because of it - maybe write some tests around the new method you added in the Config Helper?

File: docs/legal_assistance.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+## Overview
+The CWYD Legal Assistant is designed to help legal professionals efficiently manage and interact with a large collection of legal documents. It utilizes advanced natural language processing capabilities to provide accurate and contextually relevant responses to user queries about the documents.
+
+## Legal Assistant Infrastructure configuration
+
+
+The following is the CWYD infrastructure configuration that we suggest to optimize the performance and functionality of the Legal Assistant:
+
+- **Azure Semantic Search**: Utilize Azure Semantic Search to efficiently index and search legal documents. This provides powerful search capabilities and integration with other Azure services.
```
Comment (hunterjam) [inline docs/legal_assistance.md:11]: You need to specify which field to update in the deployment (if using OneClick Deploy) or the Environment Variable (if modifying an existing deployment). AZURE_SEARCH_USE_SEMANTIC_SEARCH

File: docs/legal_assistance.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+- **Azure Cognitive Search Top K 15**: Set the Top K parameter to 15 to retrieve the top 15 most relevant documents. This configuration helps in providing precise and relevant search results for user queries.
```
Comment (hunterjam) [inline docs/legal_assistance.md:13]: You need to specify which field to update in the deployment (if using OneClick Deploy) or the Environment Variable (if modifying an existing deployment). AZURE_SEARCH_TOP_K

File: docs/legal_assistance.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+- **Azure Search Integrated Vectorization**: Enable integrated vectorization in Azure Search to improve the semantic understanding and relevance of search results. This enhances the Legal Assistant's ability to provide contextually accurate answers.
```
Comment (hunterjam) [inline docs/legal_assistance.md:15]: You need to specify which field to update in the deployment (if using OneClick Deploy) or the Environment Variable (if modifying an existing deployment). AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION

File: docs/legal_assistance.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+
+- **Azure OpenAI Model gpt-4o**: Leverage the Azure OpenAI model gpt-4o for advanced natural language processing capabilities. This model is well-suited for handling complex legal queries and providing detailed and contextually appropriate responses.
```
Comment (hunterjam) [inline docs/legal_assistance.md:17]: You need to specify which field to update in the deployment (if using OneClick Deploy) or the Environment Variable (if modifying an existing deployment).  AZURE_OPENAI_MODEL / AZURE_OPENAI_MODEL_NAME

File: docs/legal_assistance.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+
+- **Azure OpenAI Model Version 2024-05-13**: Ensure that the Legal Assistant is using the latest version of the Azure OpenAI model (version 2024-05-13) to take advantage of the most recent improvements and features.
```
Comment (hunterjam) [inline docs/legal_assistance.md:19]: You need to specify which field to update in the deployment (if using OneClick Deploy) or the Environment Variable (if modifying an existing deployment). AZURE_OPENAI_API_VERSION

File: docs/legal_assistance.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+
+- **Orchestration Strategy: Semantic Kernel**: Implement the Semantic Kernel orchestration strategy to effectively manage the integration and interaction between different components of the infrastructure. This strategy ensures seamless operation and optimal performance of the Legal Assistant.
```
Comment (hunterjam) [inline docs/legal_assistance.md:21]: You need to specify which field to update in the deployment (if using OneClick Deploy) or the Environment Variable (if modifying an existing deployment). ORCHESTRATION_STRATEGY
The user can also modify this value in the Configuration settings of the Admin console.

File: docs/legal_assistance.md [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+
+- **Conversation Flow Options**: Setting Set `CONVERSATION_FLOW=byod` enables running advanced AI models like GPT-4o on your own enterprise data without needing to train or fine-tune models.
+
+- **Chunking Strategy**: Setting the chunking strategy to maximum page length size of 1200 characters to optimizing relevance, preserving context
```
Comment (hunterjam) [inline docs/legal_assistance.md:25]: Specify where the user can update this value in the Configuration settings of the Admin console.

File: docs/legal_assistance.md [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+
+![UnSelected](images/cwyd_admin_legal_unselected.png)
+
+
+- **Selected** Legal Assistant prompt
+
+![Checked ](images/cwyd_admin_legal_selected.png)
+
+
+When the user selects "Legal Assistant", the user prompt textbox will update to the Legal Assistant prompt. When the user select the default, the user prompt textbox will update to the default prompt. Note that if the user has a custom prompt in the user prompt textbox, selecting an option from the dropdown will overwrite the custom prompt with the default or legal assistant prompt.
```
Comment (hunterjam) [inline docs/legal_assistance.md:43]: Specify that the user must Save the Configuration after making this change.

**Review by hunterjam — CHANGES_REQUESTED** [review]
@Fr4nc3 @aldunson - Please address the requested changes in the PR and resubmit.

File: .gitignore (hunk 4 above)
Comment (Fr4nc3) [reply .gitignore:429]: it was added by frtibble, I removed it

File: code/backend/batch/utilities/helpers/config/config_helper.py (hunk 5 above)
Comment (Fr4nc3) [reply code/backend/batch/utilities/helpers/config/config_helper.py:198]: I think that can be both, but it should be `config["prompts"].get("ai_assistant_type")`. 
let me know if that is a good answer for you or I change the line =)

**Comment by hunterjam** [comment]
@Fr4nc3 @aldunson - The updates are looking better. However, I am unable to receive a valid response within the chat interface. I reviewed the logs and found this error. I verified that all of the endpoints and keys are correct. I deployed this solution multiple times therefore, it is repeatable.
image

**Review by hunterjam — CHANGES_REQUESTED** [review]

File: docs/legal_assistance.md [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+By following these infrastructure configurations, you can enhance the efficiency, accuracy, and overall performance of the CWYD Legal Assistant, ensuring it meets the high demands and expectations of legal professionals.
+
+## Updating Configuration Fields
+
+To apply the suggested configurations in your deployment, update the following fields accordingly:
+- **Azure Semantic Search**: Set `AZURE_SEARCH_USE_SEMANTIC_SEARCH` to `true`
+- **Azure Cognitive Search Top K 15**: Set `AZURE_SEARCH_TOP_K` to `15`.
+- **Azure Search Integrated Vectorization**: Set `AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION` to `true`.
+- **Azure OpenAI Model**: Set `AZURE_OPENAI_MODEL`  to `gpt-4o`.
+- **Azure OpenAI Model Name**: Set `AZURE_OPENAI_MODEL_NAME` to `2024-05-13`.
```
[the remark this answers is not in the document]
Comment (aldunson) [reply docs/legal_assistance.md]: done

File: docs/legal_assistance.md [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+
+By following these infrastructure configurations, you can enhance the efficiency, accuracy, and overall performance of the CWYD Legal Assistant, ensuring it meets the high demands and expectations of legal professionals.
+
+## Updating Configuration Fields
+
+To apply the suggested configurations in your deployment, update the following fields accordingly:
+- **Azure Semantic Search**: Set `AZURE_SEARCH_USE_SEMANTIC_SEARCH` to `true`
+- **Azure Cognitive Search Top K 15**: Set `AZURE_SEARCH_TOP_K` to `15`.
+- **Azure Search Integrated Vectorization**: Set `AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION` to `true`.
+- **Azure OpenAI Model**: Set `AZURE_OPENAI_MODEL`  to `gpt-4o`.
```
[the remark this answers is not in the document]
Comment (aldunson) [reply docs/legal_assistance.md]: done

File: docs/legal_assistance.md [hunk 17]
```diff
@@ -22,8 +22,8 @@ To apply the suggested configurations in your deployment, update the following f
 - **Azure Semantic Search**: Set `AZURE_SEARCH_USE_SEMANTIC_SEARCH` to `true`
 - **Azure Cognitive Search Top K 15**: Set `AZURE_SEARCH_TOP_K` to `15`.
 - **Azure Search Integrated Vectorization**: Set `AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION` to `true`.
-- **Azure OpenAI Model**: Set `AZURE_OPENAI_MODEL`  to `gpt-4o`.
-- **Azure OpenAI Model Name**: Set `AZURE_OPENAI_MODEL_NAME` to `2024-05-13`.
+- **Azure OpenAI Model Name**: Set `AZURE_OPENAI_MODEL_NAME` to `gpt-4o`. (could be different based on the name of the Azure OpenAI model deployment)
```
Comment (hunterjam) [inline docs/legal_assistance.md:25]: Should be:  
**Azure OpenAI Model**: Set `AZURE_OPENAI_MODEL` to `gpt-4o`.
**Azure OpenAI Model Name**: Set AZURE_OPENAI_MODEL_NAME to gpt-4o. (could be different based on the name of the Azure OpenAI model deployment)

**Review by hunterjam — CHANGES_REQUESTED** [review]
@aldunson - You removed the Azure OpenAI Model. The feedback was to add Azure OpenAI Model Name.

File: docs/legal_assistance.md (hunk 17 above)
Comment (aldunson) [reply docs/legal_assistance.md:26]: oops, added back.

**Review by hunterjam — APPROVED** [review]
